src/components/NodeView.tsx

```diff
@@ -16,7 +17,7 @@ export default function NodeView({ root, selectedNode, onClick }: Props) {
       className={`node-container ${selectedNode === root ? 'selected' : ''}`}
     >
       <div
-        className={`node ${selectedNode === root ? 'selected' : ''}`}
+        className={`node ${root ? root.formula : ''} ${selectedNode === root ? 'selected' : ''} `}
```

Comment:
Why add the node's formula as a class?
Is this for react-lineto?
I think you're going to run into problems if you have the same formula more than once.